src/dos.jl

```diff
@@ -0,0 +1,181 @@
+using QuadGK
+using Elliptic
+
+"""
+Descriptor of an integrable singularity in DOS
+"""
+struct DOSSingularity
+  # Position of the singularity, Ω_p.
+  position::Real
+  # Asymptotic behavior of the DOS near the singularity, S_p(ω).
+  # To be chosen such that \lim_{ω \to Ω_p} [D(ω) - S_p(ω)] = 0
+  asymptotics::Function
```

Comment:
I would remove the `::Function` annotation here so that you can use any callable object